store/sqlstore/channel_store_shared_channels.go

```diff
@@ -0,0 +1,290 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package sqlstore
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/store"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/pkg/errors"
+)
+
+// SaveSharedChannel inserts a new shared channel record.
+func (s SqlChannelStore) SaveSharedChannel(sc *model.SharedChannel) (*model.SharedChannel, error) {
+	sc.PreSave()
+	if err := sc.IsValid(); err != nil {
+		return nil, err
+	}
+
+	if err := s.GetMaster().Insert(sc); err != nil {
+		return nil, errors.Wrapf(err, "save_shared_channel: id=%s", sc.ChannelId)
+	}
+	return sc, nil
+}
+
+// GetSharedChannel fetches a shared channel by channel_id.
+func (s SqlChannelStore) GetSharedChannel(channelId string) (*model.SharedChannel, error) {
+	var sc model.SharedChannel
+
+	query := s.getQueryBuilder().
+		Select("*").
+		From("SharedChannels").
+		Where(sq.Eq{"SharedChannels.ChannelId": channelId})
+
+	squery, args, err := query.ToSql()
+	if err != nil {
+		return nil, errors.Wrapf(err, "getsharedchannel_tosql")
+	}
+
+	if err := s.GetReplica().SelectOne(&sc, squery, args...); err != nil {
+		if err == sql.ErrNoRows {
+			return nil, store.NewErrNotFound("SharedChannel", channelId)
+		}
+		return nil, errors.Wrapf(err, "failed to find channel with ChannelId=%s", channelId)
+	}
+	return &sc, nil
+}
+
+// GetSharedChannels fetches a paginated list of shared channels filtered by SharedChannelSearchOpts.
+func (s SqlChannelStore) GetSharedChannels(offset, limit int, opts store.SharedChannelFilterOpts) ([]*model.SharedChannel, error) {
```

Comment:
Just an observation - in some parts of the code we use "page" and "perPage" instead of "offset" and "limit" directly. In those places offset is calculated as `perPage * page`.

I'm not sure which one is supposed to the "correct" way to do it.

Reply:
The Rest layer (and app layer) uses page & perPage  as it makes the APIs more user friendly.  Store layer uses offset & limit because that's what the SQL uses.